models/ldndc/R/model2netcdf.LDNDC.R

```diff
@@ -33,40 +32,66 @@ model2netcdf.LDNDC <- function(outdir, sitelat, sitelon, start_date, end_date, d
+    cut <- subset(read.csv(paste(output_dir, "report-cut.txt", sep = "/"), header = T, sep ="\t"),
+                      select = c("datetime", "dC_fru_export.kgCha.1.", "dC_fol_export.kgCha.1.", "dC_dfol_export.kgCha.1.",
+                                 "dC_lst_export.kgCha.1.", "dC_dst_export.kgCha.1.", "dC_frt_export.kgCha.1."))
+    
+    cut$total <- rowSums(cut[,-1])
+    cut <- cut[,c("datetime", "total")]
+    
+  } else{
+    PEcAn.logger::logger.severe("Subdaily output files not found, check the configurations for the LDNDC runs")
   }
   
+
+  # This approach should be more reliable compared to previous since just choose one unique datetime
+  # and the last one will be the "all", if there are several species on the field
+  physiology <- physiology[!duplicated(physiology$datetime, fromLast = T),] 
   
   
-  # ldndc.out <- merge(ecosystem, physiology, by = "datetime", all = TRUE) %>%
-  #   mutate(Date = format(as.POSIXlt(datetime, format = "%Y-%m-%d")), .keep = "unused") %>%
-  #   mutate(Year = lubridate::year(Date), Day = strftime(Date, format = "%j"),
-  #          Step = rep(0:(length(Date)/length(unique(Date))-1),length(unique(Date)))) %>%
-  #   select(Year, Day, Step, dC_NEE.kgCha.1., C_total.kgCha.1., lai)
+  # Combine harvest and cut as one event
+  harvest <- rbind(harvest, cut) %>% dplyr::group_by(.data$datetime) %>% dplyr::summarise(harvest_carbon_flux = sum(.data$total)/10000) %>%
+    as.data.frame()
   
   # Temporary solution to get "no visible binding" note off from the variables: 'Date', 'Year' and 'Day'
-  Date <- Year <- Day <- NULL
+  Date <- Year <- Day <- Step <- NULL
+  
+  ## Merge subdaily-files
+  ldndc.raw.out <- merge(physiology, soilchemistry, by = 'datetime', all = TRUE)
+  ldndc.raw.out <- merge(ldndc.raw.out, watercycle, by = 'datetime', all = TRUE)
+  ldndc.raw.out <- merge(ldndc.raw.out, harvest, by = 'datetime', all = TRUE)
   
-  ldndc.out <- physiology %>%
-    dplyr:: mutate(Date = format(as.POSIXlt(.data$datetime, format = "%Y-%m-%d")), .keep = "unused") %>%
-    dplyr::slice(1:(dplyr::n()-1)) %>% # Removing one extra observation
+  ldndc.out <- ldndc.raw.out %>%
+    dplyr:: mutate(Date = format(as.POSIXct(.data$datetime, format = "%Y-%m-%d")), .keep = "unused") %>%
+    dplyr::slice(1:(dplyr::n()-1)) %>% # Removing one extra line in output
     dplyr::mutate(Year = lubridate::year(Date), Day = as.numeric(strftime(Date, format = "%j")),
            Step = rep(0:(length(which(Date %in% unique(Date)[1]))-1),len = length(Date))) %>%
-    dplyr::select('Year', 'Day', 'Step', 'lai', 'dC_co2_upt.kgCm.2.')
-  
+    dplyr::select(Year, Day, Step, .data$lai, .data$dC_maintenance_resp.kgCm.2., .data$dC_transport_resp.kgCm.2.,
+                  .data$dC_growth_resp.kgCm.2., .data$dC_co2_upt.kgCm.2., .data$sC_co2_hetero.kgCm.2.,
+                  .data$DW_below.kgDWm.2., .data$DW_above.kgDWm.2., .data$soilwater_10cm...,
+                  .data$soilwater_30cm..., .data$harvest_carbon_flux)
```

Comment:
The current Tidyverse recommendation is now to give column names as strings instead of with `.data$` -- see `vignette("in-packages", package="dplyr")`
Confusingly, this advice is only true in "tidyselect context" -- in filter and mutate `.data` is still the way to go.
```suggestion
    dplyr::select(
      "Year", "Day", "Step", "lai",
      "dC_maintenance_resp.kgCm.2.",
      "dC_transport_resp.kgCm.2.",
      "dC_growth_resp.kgCm.2.",
      "dC_co2_upt.kgCm.2.",
      "sC_co2_hetero.kgCm.2.",
      "DW_below.kgDWm.2.",
      "DW_above.kgDWm.2.",
      "soilwater_10cm...",
      "soilwater_30cm...",
      "harvest_carbon_flux")
```